### ISSUE TITLE
level_number key is not being appended to all level_completed events

### ISSUE THREAD

**Issue by burrage** [curiouslearning/FeedTheMonsterJS#979]
**User Story**
As a developer,
I want to investigate why the level_number key is not being appended to all level_completed events in FTM JS,
So that we can figure out why this discrepancy has happened.

Queries below that help describe the phenomenon. 

Looking at 1 day worth of data

```
SELECT * 
FROM `ftm-b9d99.analytics_159643920.events_20240212`
WHERE device.web_info.hostname LIKE 'feedthemonster.curiouscontent.org%' 
AND event_name = 'level_completed'
```

**692 level_completed events**

```
SELECT * 
FROM `ftm-b9d99.analytics_159643920.events_20240212`,
UNNEST(event_params) AS event_params
WHERE device.web_info.hostname LIKE 'feedthemonster.curiouscontent.org%' 
AND event_name = 'level_completed' 
AND event_params.key = 'level_number'
```

**690 events**

Why is there a 2 event discrepancy between these two queries?

Let’s look a week out to see how large this gets…

```
SELECT * 
FROM `ftm-b9d99.analytics_159643920.events_*`
WHERE
  _TABLE_SUFFIX BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 1 WEEK))
AND 
  FORMAT_DATE('%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY))  
AND device.web_info.hostname LIKE 'feedthemonster.curiouscontent.org%'
AND event_name = 'level_completed'
```

**5415 events**

```
SELECT * 
FROM `ftm-b9d99.analytics_159643920.events_*`,
UNNEST(event_params) AS event_params
WHERE
  _TABLE_SUFFIX BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 1 WEEK))
AND 
  FORMAT_DATE('%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY))  
AND device.web_info.hostname LIKE 'feedthemonster.curiouscontent.org%'
AND event_name = 'level_completed' 
AND event_params.key = 'level_number'
```

**5365 events**

**50 event difference for the last week**

Let's look at a month...

```
SELECT * 
FROM `ftm-b9d99.analytics_159643920.events_*`
WHERE
  _TABLE_SUFFIX BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 1 MONTH))
AND 
  FORMAT_DATE('%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY))  
AND device.web_info.hostname LIKE 'feedthemonster.curiouscontent.org%'
AND event_name = 'level_completed'
```

**22129 events**

```
SELECT * 
FROM `ftm-b9d99.analytics_159643920.events_*`,
UNNEST(event_params) AS event_params
WHERE
  _TABLE_SUFFIX BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 1 MONTH))
AND 
  FORMAT_DATE('%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY))  
AND device.web_info.hostname LIKE 'feedthemonster.curiouscontent.org%'
AND event_name = 'level_completed' 
AND event_params.key = 'level_number'
```

**21739 events**

**390 event difference**

**Acceptance Criteria**
Given that I have looked at the current data collection code in FTM JS,
When I look more deeply at the data where a level_number key is not appended to level_completed events,
Then I should be able to report my findings as a comment to this task card or in a Google Doc.

**Comment by ashishkrr** [comment]
Hey @burrage , we have observed from our investigation is, the missing level_number in level_completed event is only below release version 1.0.24, from release version 1.0.24 level_number is present in every level_completed event, past 3 months back we had issue with level_number in level_completed in release version 1.0.23 which was fixed in [PR](https://github.com/curiouslearning/FeedTheMonsterJS/pull/938) and this fix reflected from version 1.0.24.

**Comment by burrage** [comment]
Excellent find @ashishkrr ! 

And a good piece of data to support prioritizing that we would want to invest the time and effort to work on some QA scripts to test the structure and values of critical event data!

We can consider this one Done :-)
